index.js

```diff
@@ -346,22 +369,26 @@ const compile = function(schema, cache, root, reporter, opts) {
       consume('uniqueItems')
     }
 
-    if (node.enum) {
-      if (!Array.isArray(node.enum)) throw new Error('Invalid enum')
-
-      const complex = node.enum.some(function(e) {
-        return typeof e === 'object'
-      })
-
-      const compare = complex
-        ? function(e) {
-            return `JSON.stringify(${name})` + ` !== JSON.stringify(${JSON.stringify(e)})`
-          }
-        : function(e) {
-            return `${name} !== ${JSON.stringify(e)}`
+    const makeCompare = (name, complex) =>
+      complex
+        ? (e) => {
+            scope.deepEqual = deepEqual
+            return `!deepEqual(${name}, ${JSON.stringify(e)})`
```

Comment:
Kind of confusing that this returns an inverted result. Normally one expects comparison functions to return `true` if the operands are equal. While we have the chance, perhaps we should fix that?

Reply:
Thought about it too, but given the usage, perhaps it's simpler to rename the function instead.